pkgs/misc/vim-plugins/update.py

```diff
@@ -477,13 +482,27 @@ def parse_args():
         default=DEFAULT_OUT,
         help="Filename to save generated nix code",
     )
+    parser.add_argument(
+        "--commit",
+        dest="commit",
+        action="store_true",
+        help="Automatically commit updates",
+    )
 
     return parser.parse_args()
 
 
-def main() -> None:
+def get_nixpkgs_repo() -> git.Repo:
+    repo = git.Repo(NIXPKGS_PATH)
+    if repo.is_dirty():
+        raise Exception("Please stash your changes before updating.")
+    return repo
+
 
+def main() -> None:
```

Comment:
Is this type annotation necessary? Or is it common practice? I don't have much experience with mypy.

Reply:
No idea, but I didn't intentionally make this change so I'll revert it. This is my first experience with mypy and I must say it's been great and saved a lot of time.